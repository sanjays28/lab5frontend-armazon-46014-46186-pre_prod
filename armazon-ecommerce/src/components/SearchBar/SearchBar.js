import React, { useState, useCallback, useEffect } from 'react';
import { TextField, CircularProgress, InputAdornment, Alert } from '@mui/material';
import SearchIcon from '@mui/icons-material/Search';
import ErrorIcon from '@mui/icons-material/Error';
import debounce from 'lodash/debounce';

// PUBLIC_INTERFACE
const SearchBar = ({ onSearch, isLoading, error }) => {
  const [searchTerm, setSearchTerm] = useState('');

  // Debounce the search callback with a 500ms delay
  const debouncedSearch = useCallback(
    debounce((value) => {
      onSearch(value);
    }, 500),
    [onSearch]
  );

  // Cleanup debounced function on unmount
  useEffect(() => {
    return () => {
      debouncedSearch.cancel();
    };
  }, [debouncedSearch]);

  const handleSearchChange = (event) => {
    const value = event.target.value;
    setSearchTerm(value);
    debouncedSearch(value);
  };

  return (
    <>
    <TextField
      fullWidth
      variant="outlined"
      placeholder="Search products..."
      value={searchTerm}
      onChange={handleSearchChange}
      data-testid="search-field"
      inputProps={{
        'data-testid': 'search-input',
        role: 'textbox',
        'aria-label': 'Search products'
      }}
      InputProps={{
        startAdornment: (
          <InputAdornment position="start">
            <SearchIcon data-testid="search-icon" />
          </InputAdornment>
        ),
        endAdornment: isLoading && (
          <InputAdornment position="end">
            <CircularProgress size={20} data-testid="loading-spinner" />
          </InputAdornment>
        ),
      }}
      sx={{
        maxWidth: '600px',
        margin: '16px auto',
        '& .MuiOutlinedInput-root': {
          backgroundColor: 'background.paper',
          '&:hover': {
            '& .MuiOutlinedInput-notchedOutline': {
              borderColor: 'primary.main',
            },
          },
        },
      }}
    />
    {error && (
      <Alert 
        severity="error" 
        data-testid="search-error"
        sx={{ 
          maxWidth: '600px', 
          margin: '8px auto',
          alignItems: 'center'
        }}
        icon={<ErrorIcon />}
      >
        {error}
      </Alert>
    )}
    </>
  );
};

export default SearchBar;
