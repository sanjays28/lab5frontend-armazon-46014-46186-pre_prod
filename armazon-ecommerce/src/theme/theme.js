import { createTheme } from '@mui/material/styles';

// PUBLIC_INTERFACE
const theme = createTheme({
  palette: {
    primary: {
      main: '#1976d2',
    },
    secondary: {
      main: '#dc004e',
    },
  },
  typography: {
    fontFamily: [
      '-apple-system',
      'BlinkMacSystemFont',
      '"Segoe UI"',
      'Roboto',
      '"Helvetica Neue"',
      'Arial',
      'sans-serif',
    ].join(','),
  },
});

export default theme;