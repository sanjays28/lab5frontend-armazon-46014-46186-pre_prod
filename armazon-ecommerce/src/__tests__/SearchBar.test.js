import React from 'react';
import { render, screen, fireEvent, act, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { ThemeProvider } from '@mui/material/styles';
import renderer from 'react-test-renderer';
import SearchBar from '../components/SearchBar/SearchBar';
import theme from '../theme/theme';

// Helper function to wait for Material-UI animations
const waitForMuiAnimation = () => new Promise(resolve => setTimeout(resolve, 0));

// Custom render function that includes Material-UI ThemeProvider
const customRender = (ui, options) =>
  render(ui, {
    wrapper: ({ children }) => (
      <ThemeProvider theme={theme}>
        {children}
      </ThemeProvider>
    ),
    ...options,
  });

// Mock lodash debounce to make it synchronous for testing
jest.mock('lodash/debounce', () => {
  return jest.fn(fn => {
    const debounced = (...args) => fn.apply(null, args);
    debounced.cancel = jest.fn();
    return debounced;
  });
});

// Increase test timeout
jest.setTimeout(10000);

// Snapshot Tests
describe('Snapshot Tests', () => {
  const renderWithTheme = async (component) => {
    let tree;
    await act(async () => {
      tree = renderer.create(
        <ThemeProvider theme={theme}>
          {component}
        </ThemeProvider>
      );
      await waitForMuiAnimation();
    });
    return tree;
  };

  test('matches snapshot in default state', async () => {
    const tree = await renderWithTheme(
      <SearchBar onSearch={() => {}} isLoading={false} />
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });

  test('matches snapshot in loading state', async () => {
    const tree = await renderWithTheme(
      <SearchBar onSearch={() => {}} isLoading={true} />
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });

  test('matches snapshot with search text', async () => {
    const tree = await renderWithTheme(
      <SearchBar onSearch={() => {}} isLoading={false} />
    );
    await act(async () => {
      const instance = tree.root;
      const input = instance.findByProps({ 'data-testid': 'search-input' });
      input.props.onChange({ target: { value: 'test search' } });
      await waitForMuiAnimation();
    });
    expect(tree.toJSON()).toMatchSnapshot();
  });

  test('matches snapshot with different screen sizes', async () => {
    const originalMatchMedia = window.matchMedia;
    const mockMatchMedia = (query) => ({
      matches: query === '(min-width: 600px)',
      media: query,
      onchange: null,
      addListener: jest.fn(),
      removeListener: jest.fn(),
      addEventListener: jest.fn(),
      removeEventListener: jest.fn(),
      dispatchEvent: jest.fn(),
    });

    window.matchMedia = mockMatchMedia;
    const tree = await renderWithTheme(
      <SearchBar onSearch={() => {}} isLoading={false} />
    );
    expect(tree.toJSON()).toMatchSnapshot('desktop view');

    window.matchMedia = (query) => ({
      ...mockMatchMedia(query),
      matches: false,
    });
    const mobileTree = await renderWithTheme(
      <SearchBar onSearch={() => {}} isLoading={false} />
    );
    expect(mobileTree.toJSON()).toMatchSnapshot('mobile view');

    window.matchMedia = originalMatchMedia;
  });
});


describe('SearchBar Component', () => {
  beforeEach(() => {
    jest.clearAllMocks();
    jest.useFakeTimers();
  });

  afterEach(() => {
    jest.clearAllTimers();
    jest.useRealTimers();
    jest.clearAllMocks();
  });

  // Test 1: Basic Rendering
  test('renders search input field with placeholder', () => {
    customRender(<SearchBar onSearch={() => {}} isLoading={false} />);
    expect(screen.getByTestId('search-input')).toBeInTheDocument();
    expect(screen.getByTestId('search-input')).toHaveAttribute('placeholder', 'Search products...');
  });

  // Test 2: Search Icon Rendering
  test('displays search icon', () => {
    customRender(<SearchBar onSearch={() => {}} isLoading={false} />);
    expect(screen.getByTestId('search-icon')).toBeInTheDocument();
  });

  // Test 3: Loading State
  test('shows loading indicator when isLoading is true', () => {
    customRender(<SearchBar onSearch={() => {}} isLoading={true} />);
    expect(screen.getByTestId('loading-spinner')).toBeInTheDocument();
  });

  // Test 4: No Loading Indicator
  test('does not show loading indicator when isLoading is false', () => {
    customRender(<SearchBar onSearch={() => {}} isLoading={false} />);
    expect(screen.queryByTestId('loading-spinner')).not.toBeInTheDocument();
  });

  // Test 5: Input Change Handler
  test('calls onSearch with input value', async () => {
    const mockOnSearch = jest.fn();
    customRender(<SearchBar onSearch={mockOnSearch} isLoading={false} />);
    
    const searchInput = screen.getByTestId('search-input');
    
    await act(async () => {
      await userEvent.type(searchInput, 'test');
    });
    
    expect(mockOnSearch).toHaveBeenCalledWith('test');
  });

  // Test 6: Debounce Functionality
  test('debounces search callback', async () => {
    const mockOnSearch = jest.fn();
    customRender(<SearchBar onSearch={mockOnSearch} isLoading={false} />);
    
    const searchInput = screen.getByTestId('search-input');
    
    // Type rapidly
    await act(async () => {
      await userEvent.type(searchInput, 'test');
    });
    
    // Verify the call was made
    expect(mockOnSearch).toHaveBeenCalledTimes(4); // Once for each character
    expect(mockOnSearch).toHaveBeenLastCalledWith('test');
  });

  // Test 7: Empty Input Handling
  test('calls onSearch with empty string when input is cleared', async () => {
    const mockOnSearch = jest.fn();
    customRender(<SearchBar onSearch={mockOnSearch} isLoading={false} />);
    
    const searchInput = screen.getByTestId('search-input');
    
    await act(async () => {
      await userEvent.type(searchInput, 'test');
      await userEvent.clear(searchInput);
    });
    
    expect(mockOnSearch).toHaveBeenCalledWith('');
  });

  // Test 8: Input State Management
  test('maintains input value state', async () => {
    customRender(<SearchBar onSearch={() => {}} isLoading={false} />);
    
    const searchInput = screen.getByTestId('search-input');
    await userEvent.type(searchInput, 'test value');
    
    expect(searchInput).toHaveValue('test value');
  });

  // Test 9: Special Characters Handling
  test('handles special characters in search input', async () => {
    const mockOnSearch = jest.fn();
    customRender(<SearchBar onSearch={mockOnSearch} isLoading={false} />);
    
    const searchInput = screen.getByTestId('search-input');
    await userEvent.type(searchInput, '!@#$%^&*()');
    
    expect(mockOnSearch).toHaveBeenCalledWith('!@#$%^&*()');
  });

  // Test 10: Long Input Handling
  test('handles long search queries', async () => {
    const mockOnSearch = jest.fn();
    const longString = 'a'.repeat(100);
    customRender(<SearchBar onSearch={mockOnSearch} isLoading={false} />);
    
    const searchInput = screen.getByTestId('search-input');
    await userEvent.type(searchInput, longString);
    
    expect(mockOnSearch).toHaveBeenCalledWith(longString);
  });

  // Test 11: Rapid Input Changes
  test('handles rapid input changes correctly', async () => {
    const mockOnSearch = jest.fn();
    customRender(<SearchBar onSearch={mockOnSearch} isLoading={false} />);
    
    const searchInput = screen.getByTestId('search-input');
    
    await act(async () => {
      await userEvent.type(searchInput, 'test');
      await userEvent.clear(searchInput);
      await userEvent.type(searchInput, 'new');
    });
    
    expect(mockOnSearch).toHaveBeenLastCalledWith('new');
    expect(mockOnSearch).toHaveBeenCalledTimes(8); // 4 for 'test' + 1 for clear + 3 for 'new'
  });

  // Test 12: Component Props Update
  test('updates when isLoading prop changes', () => {
    const { rerender } = customRender(<SearchBar onSearch={() => {}} isLoading={false} />);
    expect(screen.queryByTestId('loading-spinner')).not.toBeInTheDocument();
    
    rerender(<SearchBar onSearch={() => {}} isLoading={true} />);
    expect(screen.getByTestId('loading-spinner')).toBeInTheDocument();
  });

  // Test 13: Input Accessibility
  test('search input is accessible', () => {
    customRender(<SearchBar onSearch={() => {}} isLoading={false} />);
    const searchInput = screen.getByTestId('search-input');
    
    expect(searchInput).toHaveAttribute('role', 'textbox');
    expect(searchInput).not.toHaveAttribute('aria-disabled');
  });

  // Test 14: Error Display
  test('displays error message when error prop is provided', () => {
    const errorMessage = 'Failed to search products';
    customRender(<SearchBar onSearch={() => {}} isLoading={false} error={errorMessage} />);
    
    const errorAlert = screen.getByTestId('search-error');
    expect(errorAlert).toBeInTheDocument();
    expect(errorAlert).toHaveTextContent(errorMessage);
  });

  // Test 15: No Error Display
  test('does not display error message when error prop is null', () => {
    customRender(<SearchBar onSearch={() => {}} isLoading={false} error={null} />);
    expect(screen.queryByTestId('search-error')).not.toBeInTheDocument();
  });

  // Test 16: Error Cleanup
  test('removes error message when error prop changes to null', () => {
    const { rerender } = customRender(
      <SearchBar onSearch={() => {}} isLoading={false} error="Error message" />
    );
    
    expect(screen.getByTestId('search-error')).toBeInTheDocument();
    
    rerender(<SearchBar onSearch={() => {}} isLoading={false} error={null} />);
    expect(screen.queryByTestId('search-error')).not.toBeInTheDocument();
  });

  // Test 17: Debounce Cleanup
  test('cleans up debounced function on unmount', () => {
    const mockOnSearch = jest.fn();
    const { unmount } = customRender(<SearchBar onSearch={mockOnSearch} isLoading={false} />);
    
    const searchInput = screen.getByTestId('search-input');
    fireEvent.change(searchInput, { target: { value: 'test' } });
    
    unmount();
    
    // After unmount, no new calls should be made
    jest.advanceTimersByTime(1000);
    expect(mockOnSearch).toHaveBeenCalledTimes(1);
  });
});
