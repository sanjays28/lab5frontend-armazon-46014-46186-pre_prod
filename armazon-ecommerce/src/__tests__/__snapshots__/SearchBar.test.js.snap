// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Tests matches snapshot in default state 1`] = `
<div
  className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-oievqz-MuiFormControl-root-MuiTextField-root"
  data-testid="search-field"
>
  <div
    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedStart css-13w5xxm-MuiInputBase-root-MuiOutlinedInput-root"
    onClick={[Function]}
  >
    <div
      className="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-ittuaa-MuiInputAdornment-root"
    >
      <span
        className="notranslate"
      >
        ​
      </span>
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="search-icon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
        />
      </svg>
    </div>
    <input
      aria-invalid={false}
      aria-label="Search products"
      autoFocus={false}
      className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart css-1o9s3wi-MuiInputBase-input-MuiOutlinedInput-input"
      data-testid="search-input"
      disabled={false}
      id=":r0:"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Search products..."
      required={false}
      role="textbox"
      type="text"
      value=""
    />
    <fieldset
      aria-hidden={true}
      className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="css-ihdtdm"
      >
        <span
          className="notranslate"
        >
          ​
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`Snapshot Tests matches snapshot in loading state 1`] = `
<div
  className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-oievqz-MuiFormControl-root-MuiTextField-root"
  data-testid="search-field"
>
  <div
    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedStart MuiInputBase-adornedEnd css-1ew9v79-MuiInputBase-root-MuiOutlinedInput-root"
    onClick={[Function]}
  >
    <div
      className="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-ittuaa-MuiInputAdornment-root"
    >
      <span
        className="notranslate"
      >
        ​
      </span>
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="search-icon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
        />
      </svg>
    </div>
    <input
      aria-invalid={false}
      aria-label="Search products"
      autoFocus={false}
      className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart MuiInputBase-inputAdornedEnd css-152mnda-MuiInputBase-input-MuiOutlinedInput-input"
      data-testid="search-input"
      disabled={false}
      id=":r1:"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Search products..."
      required={false}
      role="textbox"
      type="text"
      value=""
    />
    <div
      className="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
    >
      <span
        className="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-18lrjg1-MuiCircularProgress-root"
        data-testid="loading-spinner"
        role="progressbar"
        style={
          Object {
            "height": 20,
            "width": 20,
          }
        }
      >
        <svg
          className="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
            cx={44}
            cy={44}
            fill="none"
            r={20.2}
            strokeWidth={3.6}
            style={Object {}}
          />
        </svg>
      </span>
    </div>
    <fieldset
      aria-hidden={true}
      className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="css-ihdtdm"
      >
        <span
          className="notranslate"
        >
          ​
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`Snapshot Tests matches snapshot with different screen sizes: desktop view 1`] = `
<div
  className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-oievqz-MuiFormControl-root-MuiTextField-root"
  data-testid="search-field"
>
  <div
    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedStart css-13w5xxm-MuiInputBase-root-MuiOutlinedInput-root"
    onClick={[Function]}
  >
    <div
      className="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-ittuaa-MuiInputAdornment-root"
    >
      <span
        className="notranslate"
      >
        ​
      </span>
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="search-icon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
        />
      </svg>
    </div>
    <input
      aria-invalid={false}
      aria-label="Search products"
      autoFocus={false}
      className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart css-1o9s3wi-MuiInputBase-input-MuiOutlinedInput-input"
      data-testid="search-input"
      disabled={false}
      id=":r3:"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Search products..."
      required={false}
      role="textbox"
      type="text"
      value=""
    />
    <fieldset
      aria-hidden={true}
      className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="css-ihdtdm"
      >
        <span
          className="notranslate"
        >
          ​
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`Snapshot Tests matches snapshot with different screen sizes: mobile view 1`] = `
<div
  className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-oievqz-MuiFormControl-root-MuiTextField-root"
  data-testid="search-field"
>
  <div
    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedStart css-13w5xxm-MuiInputBase-root-MuiOutlinedInput-root"
    onClick={[Function]}
  >
    <div
      className="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-ittuaa-MuiInputAdornment-root"
    >
      <span
        className="notranslate"
      >
        ​
      </span>
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="search-icon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
        />
      </svg>
    </div>
    <input
      aria-invalid={false}
      aria-label="Search products"
      autoFocus={false}
      className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart css-1o9s3wi-MuiInputBase-input-MuiOutlinedInput-input"
      data-testid="search-input"
      disabled={false}
      id=":r4:"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Search products..."
      required={false}
      role="textbox"
      type="text"
      value=""
    />
    <fieldset
      aria-hidden={true}
      className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="css-ihdtdm"
      >
        <span
          className="notranslate"
        >
          ​
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`Snapshot Tests matches snapshot with search text 1`] = `
<div
  className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-oievqz-MuiFormControl-root-MuiTextField-root"
>
  <div
    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedStart css-8j6b76-MuiInputBase-root-MuiOutlinedInput-root"
    onClick={[Function]}
  >
    <div
      className="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-ittuaa-MuiInputAdornment-root"
    >
      <span
        className="notranslate"
      >
        ​
      </span>
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="search-icon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
        />
      </svg>
    </div>
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart css-1o9s3wi-MuiInputBase-input-MuiOutlinedInput-input"
      data-testid="search-input"
      disabled={false}
      id=":r2:"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Search products..."
      required={false}
      role="textbox"
      type="text"
      value="test search"
    />
    <fieldset
      aria-hidden={true}
      className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="css-ihdtdm"
      >
        <span
          className="notranslate"
        >
          ​
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;
